src/PostgREST/Request/ApiRequest.hs

```diff
@@ -94,6 +94,16 @@ data Action = ActionCreate       | ActionRead{isHead :: Bool}
             | ActionSingleUpsert | ActionInvoke InvokeMethod
             | ActionInfo         | ActionInspect{isHead :: Bool}
             deriving Eq
+-- | The path info that will be mapped to a target (used to handle validations and errors before defining the Target)
+data Path
+  = PathInfo
+      { pSchema        :: Schema,
+        pName          :: Text,
+        pHasRpc        :: Bool,
+        pIsDefaultSpec :: Bool,
+        pIsRootSpec    :: Bool
+      }
+  | PathUnknown
```

Comment:
This type does still carry almost the same info as our Target type, however I'm not seeing a way to refactor it for now; so it's good to leave it like that.

The whole `ApiRequest` module needs a big refactor, it has turned messy over the years.